src/cascadia/TerminalSettingsEditor/NewTabMenuViewModel.cpp

```diff
@@ -452,6 +452,24 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         return _folderTreeCache;
     }
 
+    Collections::IObservableVector<Editor::FolderEntryViewModel> NewTabMenuViewModel::FolderTreeFlatList() const
+    {
+        std::vector<Editor::FolderEntryViewModel> flatList;
+        std::function<void(const IVector<Editor::NewTabMenuEntryViewModel>&)> addFoldersRecursively;
+        addFoldersRecursively = [&flatList, &addFoldersRecursively](const IVector<Editor::NewTabMenuEntryViewModel>& entries) {

```

Comment:
If you pass `flatList` as a function argument, you can make this a regular free standing function (`static void` or something).